actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -106,11 +106,11 @@ def decode_credentials(request)
       end
 
       def auth_scheme(request)
-        request.authorization.split(' ', 2).first
+        (request.authorization || '').split(' ', 2).first
```

Comment:
Can do `request.authorization.to_s.split('  ', 2).first`

`.to_s` will not make a new object if orginal is already a string, so shouldn't be a performance issue.